### PR TITLE
Shepherdly yml config

### PR DESCRIPTION
**Pull request by mgreene**
Shepherdly now supports `yml` config at the repo level as well as custom mitigations.

Another notable change is the support of custom mitigations. Per @def-'s [request](https://shepherdly.slack.com/archives/C051BG4BFL7/p1704815224163999), I've added a custom mitigation that links out to running nightly tests. 

Everything else in the file is a copy of how the repo settings were configured to date.

### Motivation
Shepherdly is moving towards injectable configuration and this change brings the `materialize` repo up to parity.

### CODE REVIEW

**Comment by benesch** [comment]
@def- deferring approval here to you!

File: .config/shepherdly.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  - name: unit_test
+    # A value of 100 effectively makes this always optional
+    score_threshold: 100
+  - name: observability
+    score_threshold: 75
+  - name: run_nightly_tests
+    score_threshold: 80
+    custom: true
+    label: Run Nightly Tests
+    label_url: https://trigger-ci.dev.materialize.com/?pr={{pr_number}}
```
Comment (benesch) [inline .config/shepherdly.yml:43]: Exciting to see this level of configurability!

**Comment by def-** [comment]
@mgreene Is this supposed to work already or did we do something wrong? I expected the "Run Nightly Tests" in https://github.com/MaterializeInc/materialize/pull/24355#issuecomment-1894595762 , but it didn't appear.

**Comment by mgreene** [comment]
> @mgreene Is this supposed to work already or did we do something wrong? I expected the "Run Nightly Tests" in [#24355 (comment)](https://github.com/MaterializeInc/materialize/pull/24355#issuecomment-1894595762) , but it didn't appear.

When `mitigation` configs are changed, they take effect on newly created PRs that are opened after `yml` was merged.

**Pull request merged**